include/core/CConcurrentWrapper.h

```diff
@@ -30,24 +30,25 @@ namespace core {
 //! See https://channel9.msdn.com/Shows/Going+Deep/C-and-Beyond-2012-Herb-Sutter-Concurrency-and-Parallelism
 //!
 //! @tparam T the wrapped object
-//! @tparam QUEUE_CAPACITY internal queue capacity
-//! @tparam NOTIFY_CAPACITY special parameter, for signaling the producer in blocking case
-template<typename T, size_t QUEUE_CAPACITY = 100, size_t NOTIFY_CAPACITY = 50>
+template<typename T>
 class CConcurrentWrapper final : private CNonCopyable {
 public:
     //! Wrap and return the wrapped object
     //!
     //! The object has to wrapped once and only once, pass the reference around in your code.
     //! This starts a background thread.
-    explicit CConcurrentWrapper(T& resource)
-        : m_Resource(resource), m_Done(false) {
+    explicit CConcurrentWrapper(T& resource, std::size_t queueCapacity, std::size_t notifyCapacity)
```

Comment:
How about just defaulting parameters here and ditching extra constructor:
```suggestion
    explicit CConcurrentWrapper(T& resource, std::size_t queueCapacity = 100, std::size_t notifyCapacity = 50)
```